nva/nvafakebios.c

```diff
@@ -226,14 +228,20 @@ int main(int argc, char **argv) {
 	}
 
 	/* Arguments parsing */
-	while ((c = getopt (argc, argv, "hc:e:E:w:W:l:L:")) != -1)
+	while ((c = getopt (argc, argv, "hc:m:e:E:w:W:l:L:")) != -1)
 		switch (c) {
 			case 'h':
 				usage(0);
 				break;
 			case 'c':
 				sscanf(optarg, "%d", &cnum);
 				break;
+			case 'm':
+				sscanf(optarg, "%d", &vram_size);
+				/* Calculate vbios vram location by vram size in MB */
+				vbios_vram = vram_size * 0x10 - 1;
+				vbios_vram = (vbios_vram << 12) + 0xe09;
```

Comment:
Can you explain this, at least in the commit message, if not as a comment here? What is this calculation trying to achieve?

Also, the whole point of reading the location from 0x1700/0x619f04 is so that the blob (and nouveau) would read it from there on load. Presumably when this option is taken, you should proactively write to those locations the new address you've computed?

Reply:
both registers are essentially empty on non main GPUs.

Reply:
ohh, and we write to 0x1700 already in line 90

Reply:
and restore old_bar0_pramin a short while later... 1700 is not the right place though, 619f04 is.

and what's the e09 offset?

Reply:
0x09 is SYSTEM_RAM + ENABLED
the e is just for setting the physical address at the end of the VRAM... I am not entirely sure where this VRAM_SIZE - 128kb thing comes from, but it seemed to work and is usually the value you find inside 0x619f04 if your GPU is the main one and nvidia POSTed the device or something.

Reply:
but yeah, I would prefer a way to calculate this value, which is more self explaining on what is happening there + a better comment on this as well.

Reply:
I am rewriting it. The approach to calculate the address is not logical and it will be different.